eth2util/keystore/keystore.go

```diff
@@ -62,6 +67,72 @@ func StoreKeys(secrets []*bls_sig.SecretKey, dir string) error {
 	return storeKeysInternal(secrets, dir, "keystore-%d.json")
 }
 
+// keymanagerReq represents the keymanager API request body for POST request. Refer: https://ethereum.github.io/keymanager-APIs/#/Local%20Key%20Manager/importKeystores
+type keymanagerReq struct {
+	Keystores []keystore `json:"keystores"`
+	Passwords []string   `json:"passwords"`
+}
+
+// keymanagerReqBody converts the provided secrets into keymanagerReq and returns the marshalled request.
+func keymanagerReqBody(secrets []*bls_sig.SecretKey) ([]byte, error) {
+	var req keymanagerReq
+	for _, secret := range secrets {
+		password, err := randomHex32()
+		if err != nil {
+			return nil, err
+		}
+
+		store, err := encrypt(secret, password, rand.Reader)
+		if err != nil {
+			return nil, err
+		}
+
+		req.Keystores = append(req.Keystores, store)
+		req.Passwords = append(req.Passwords, password)
+	}
+
+	res, err := json.Marshal(req)
+	if err != nil {
+		return nil, errors.New("marshal keymanager request body")
+	}
+
+	return res, nil
+}
+
+// PostKeysToKeymanager pushes the secrets to the provided keymanager address. The HTTP request times out after 2s.
+func PostKeysToKeymanager(ctx context.Context, addr string, secrets []*bls_sig.SecretKey) error {
```

Comment:
libraries shouldn't do IO, libaries should do logic and data, rather return the request type and let the user do IO.

Reply:
By library, you mean the `keystore` package here ?

Reply:
yes